dvc/remote/base.py

```diff
@@ -463,14 +450,27 @@ def create_taskset(amount):
 
     @index_locked
     def push(self, cache, named_cache, jobs=None, show_checksums=False):
-        return self._process(
+        ret = self._process(
             cache,
             named_cache,
             jobs=jobs,
             show_checksums=show_checksums,
             download=False,
         )
 
+        if self.tree.scheme == "local":
+            with self.tree.state:
+                for checksum in named_cache.scheme_keys("local"):
+                    cache_file = self.tree.hash_to_path_info(checksum)
+                    if self.tree.exists(cache_file):
+                        hash_info = HashInfo(
+                            self.tree.PARAM_CHECKSUM, checksum
+                        )
+                        self.tree.state.save(cache_file, hash_info)
+                        self.cache.protect(cache_file)
```

Comment:
Same as "trust" logic in `pull`. This really emphasizes that push and pull are the same thing: transering objects from one cache to another, so the receiving cache should be able to save objects as it pleases. This and the "trust" logic below will move to cache.save() in the future.